pkg/healthcheck/healthcheck.go

```diff
@@ -404,6 +418,9 @@ func (hc *HealthChecker) allCategories() []category {
 					description: "control plane namespace does not already exist",
 					hintAnchor:  "pre-ns",
 					check: func(context.Context) error {
+						if hc.NoInitContainer {
```

Comment:
Yes this is correct. We can have a separate namespace for the CNI and that would not cause any major trouble. It will also simplify the logic right here.  Currently the `linkerd install-cni` uses the control plane namespace that is set as a flag.

Reply:
seems to me that this is our opportunity to change this before we take off the experimental flag.  I don't have any first hand experience with the install cni flow, but it seems like putting it in its own namespace by default (e.g. `linkerd-cni`) would simplify things.

Reply:
I also wouldn't complain if we renamed NoInitContainer to CniEnabled.

Reply:
Is it possible to separate the CNI check from the control plane pre checks?  IIUC, the way that this PR implements the CNI check is by using `linkerd check --pre --linkerd-cni-enabled`. The `healthcheck. LinkerdPreInstallChecks` check category will always be run, even though it has nothing to do with CNI installation.

What I mean is:

```
linkerd install-cni | k apply -f -
linkerd check --cni # only run checks under the healthcheck.LinkerdCniPluginChecks catgory
linkerd check --pre # work as before
linkerd install | k apply -f -
etc...
```

Reply:
> I also wouldn't complain if we renamed NoInitContainer to CniEnabled.

Agreed! I created #3930 for that and will do it, just dont want to stuff it into this PR as it will make reviewing harder.

Reply:
Yes I had it like that initially but @adleong comented that we can get rid of that flag and just have it run as part of the --pre checks. Do not reall have a strong opinion on that one. The way I see it is that` linkerd check --pre --linkerd-cni-enabled `pretty much shows whether you can install linkerd with cni enabled, which really consists of all the usual checks + the checks that ensure CNI is installed and in good shape...

Reply:
The goal of #3804 is to answer the question, "_did my CNI install correctly?_" I didn't expect `--pre` or anything else to change, [such as adding the CNI checks in the multi-stage check](https://github.com/linkerd/linkerd2/blob/748b026d31c3c4c0ebb683b7b9c977e0b4fe5ddd/cli/cmd/check.go#L176). fwiw, the person running `linkerd install-cni` may not be the same person who runs `linkerd check --pre`. IOW, just one command to tell me whether my CNI is healthy.

 Hopefully, this change in direction didn't complicate the implementation details. If it does, we should re-evaluate where we want to go with this. Thanks!

Reply:
I don't necessarily have all the context, so I apologize if I'm missing the intent behind the original issue.  It seems to me like it makes sense to combine `linkerd check --cni` and `linkerd check --pre` into a single step which checks both.  In my mind, when you're asking the question "did my CNI install correctly?" what you *really* want to know is "is it okay to proceed with installing Linkerd?" and having the CNI correctly installed is just one part of that.  Therefore, I see the workflow as being:

```
linkerd install-cni | k apply -f -
linkerd check --pre --linkerd-cni-enabled
linkerd install --linkerd-cni-enabled | k apply -f -
linkerd check
```

Reply:
The CNI health check is useful beyond the installation flow. AFAICT from this PR, the CNI check is run during:

```
linkerd check --pre --linkerd-cni-enabled
linkerd check
linkerd check --proxy
```

which is fine I guess? Personally, I still prefer a simpler one CNI command (like `linkerd check --cni`), with everything else control plane-related remains unchanged. At this point, I lean towards whichever introduces less complexity into the code. Thanks.